pandas/core/ops/common.py

```diff
@@ -62,6 +65,25 @@ def new_method(self, other):
 
         other = item_from_zerodim(other)
 
+        if isinstance(self, (ABCSeries, ABCDataFrame)) and isinstance(
+            other, (ABCSeries, ABCDataFrame)
+        ):
+            # we dont require length matches
+            pass
+        elif is_list_like(other, allow_sets=False) and not isinstance(
+            other, (dict, UserDict)
```

Comment:
whey do you need UserDict here at all?

Reply:
JSONArray tests fail without this; i agree its ugly